### PR TITLE
[WIP] npm package

### PR DESCRIPTION
**Pull request by wngr**

### CODE REVIEW

File: api/src/lib.rs [hunk 1]
```diff
@@ -136,6 +137,50 @@ impl Doc {
...
+                    tlfs::PrimitiveKind::Bool => "Reg<bool>",
+                    tlfs::PrimitiveKind::U64 => "Reg<u64>",
+                    tlfs::PrimitiveKind::I64 => "Reg<i64>",
+                    tlfs::PrimitiveKind::Str => "Reg<string>",
+                }
+                .into(),
+                ArchivedSchema::Table(_, _)
+                | ArchivedSchema::Array(_)
+                | ArchivedSchema::Struct(_) => unreachable!(),
+            })
```
Comment (dvc94ch) [inline api/src/lib.rs:162]: can't we just have one api called type_of that returns an Option<String>?

**Review by dvc94ch — APPROVED** [review]
looking good, quite neat!

File: src/transport.rs [hunk 2]
```diff
@@ -16,16 +16,20 @@ fn native_transport(keypair: identity::Keypair) -> Result<Boxed<(PeerId, StreamM
...
         tcp::TcpConfig,
         yamux::YamuxConfig,
         Transport,
     };
+    use libp2p_webrtc::WebRtcTransport;
 
+    let peer_id = PeerId::from(keypair.public());
+    let webrtc = WebRtcTransport::new(peer_id, vec!["stun:stun.l.google.com:19302"]);
     let tcp = TcpConfig::new().nodelay(true);
+    let transport = core::transport::OrTransport::new(webrtc, tcp);
```
Comment (rklaehn) [inline src/transport.rs:30]: So the WebRTC transport is enabled even for non wasm builds?

**Review by rklaehn — APPROVED** [review]
Looks good, but I did not really look at the js part in detail. Would love to try it out.

File: src/transport.rs (hunk 2 above)
Comment (dvc94ch) [reply src/transport.rs]: webrtc is needed to connect to a browser node and to the cloud-relay which let's us communicate when we aren't in the same local network

File: src/transport.rs [hunk 3]
```diff
@@ -27,7 +28,8 @@ fn native_transport(keypair: identity::Keypair) -> Result<Boxed<(PeerId, StreamM
     let peer_id = PeerId::from(keypair.public());
     let webrtc = WebRtcTransport::new(peer_id, vec!["stun:stun.l.google.com:19302"]);
     let tcp = TcpConfig::new().nodelay(true);
-    let transport = core::transport::OrTransport::new(webrtc, tcp);
+    let dns = TokioDnsConfig::custom(tcp, ResolverConfig::cloudflare(), Default::default())?;
+    let transport = core::transport::OrTransport::new(webrtc, dns);
     let key = noise::Keypair::<X25519Spec>::new().into_authentic(&keypair)?;
```
Comment (dvc94ch) [inline src/transport.rs:33]: Isn't this a regression on android?

Comment (wngr) [reply src/transport.rs:33]: This should work with the custom `ResolverConfig::cloudflare()`, so no system lookup is performed. Should work fine?

File: src/transport.rs (hunk 2 above)
Comment (wngr) [reply src/transport.rs]: Exactly, so there can be direct connections native<->browser

**Comment by wngr** [comment]
As discussed, let's merge this and I will create another PR with fixes and cleanups.

**Pull request merged**